src/detector/detector.py

```diff
@@ -1,3 +1,206 @@
+            )
+
+        logger.debug("Received consumer message as json data.")
+        logger.debug(f"(data={self.messages})")
+
+    def _sha256sum(self, file_path: str) -> str:
+        """Return a SHA265 sum check to validate the model.
+
+        Args:
+            file_path (str): File path of model.
+
+        Returns:
+            str: SHA256 sum
+        """
+        h = hashlib.sha256()
+
+        with open(file_path, "rb") as file:
+            while True:
+                # Reading is buffered, so we can read smaller chunks.
+                chunk = file.read(h.block_size)
+                if not chunk:
+                    break
+                h.update(chunk)
+
+        return h.hexdigest()
+
+    def _get_model(self) -> None:
+        """Downloads model from server. If model already exists, it returns the current model. In addition, it checks the sha256 sum in case a model has been updated."""
+
+        if not os.path.isfile(self.model_path):
+
+            response = requests.get(
+                f"{MODEL_BASE_URL}/files/?p=%2F{MODEL}_{CHECKSUM}.pkl&dl=1"
+            )
+            logger.info(f"{MODEL_BASE_URL}/files/?p=%2F{MODEL}_{CHECKSUM}.pkl&dl=1")
+            response.raise_for_status()
+
+            with open(self.model_path, "wb") as f:
+                f.write(response.content)
+
+        # Check file sha256
+        local_checksum = self._sha256sum(self.model_path)
+
+        if local_checksum != CHECKSUM:
+            logger.warning(
+                f"Checksum {CHECKSUM} SHA256 is not equal with new checksum {local_checksum}!"
+            )
+            raise WrongChecksum(
+                f"Checksum {CHECKSUM} SHA256 is not equal with new checksum {local_checksum}!"
+            )
+
+        return joblib.load(self.model_path)
+
+    def clear_data(self):
+        """Clears the data in the internal data structures."""
+        self.messages = []
+        self.begin_timestamp = None
+        self.end_timestamp = None
+        self.warnings = []
+
+    def detect(self) -> None:  # pragma: no cover
+        for message in self.messages:
+            y_pred = self.model.predict(message["host_domain_name"])
+            if y_pred > THRESHOLD:
+                self.warnings.append(message)
+
+    def send_warning(self) -> None:
+        with open(f"{tempfile.gettempdir()}/warnings.json", "a") as f:
+            json.dump(self.messages, f)
+            f.write("\n")
+
+
+def main(one_iteration: bool = False):  # pragma: no cover
+    """
+    Creates the :class:`Detector` instance. Starts a loop that continously fetches data.
+
+    Args:
+        one_iteration (bool): For testing purposes: stops loop after one iteration
+
+    Raises:
+        KeyboardInterrupt: Execution interrupted by user. Closes down the :class:`LogCollector` instance.
+    """
+    logger.info("Starting Inspector...")
+    detector = Detector()
+    logger.info(f"Inspector is running.")
```

Comment:
Same as above (``Detector is running.``, ``f`` can also be removed)

Reply:
Crtl + C, Crtl + V error :smile: